experimental/dispatch_profiler/profiler.py

```diff
@@ -0,0 +1,168 @@
+
+  parser.add_argument("--tags", default='', \
+                      help="Inserts leading columns in output table and uniform "\
+                        "values for each column. Useful for generating pivot tables.")
+
+  # Parse the command line arguments.
+  args = parser.parse_args()
+  ###############################################################################
+
+  # Boolenize the string arguments from command line.
+  verification_enabled = False if args.verification_enabled in [
+      'False', 'false', '0'
+  ] else True
+  profiling_enabled = False if args.profiling_enabled in [
+      'False', 'false', '0'
+  ] else True
+  compile_only = False if args.compile_only in ['False', 'false', '0'] else True
+  # Overrite verification and profiling if compile_only is set.
+  if compile_only:
+    verification_enabled = False
+    profiling_enabled = False
+
+  # Manifests metadata for a group of accompanying operations and configurations.
+  manifest = Manifest(args)
+
+  # Collect all the avialable operations in a manifest.
+  gpu_matmul_tensor_cores_f16(manifest)
+  gpu_matmul_tensor_cores_f32(manifest)
+
+  # Performance report
+  perf_report = PerformanceReport(args)
+
+  # For all the operations in the manifest compile, verify, and profile.
+  for operation_kind, operation_collection_list in manifest.operations.items():
+    for operation_collection in operation_collection_list:
+
+      # Select and create an instance of operation_launcher for the operation with operation_kind.
+      # print(operation_collection.operation.name())
+      operation_launcher = operation_launcher_map[operation_kind](
+          args, operation_collection.operation)
+
+      for configuration in operation_collection.configuration_list:
+
+        # Compile the operation dispatches for verification and profiling.
+        if compile_only:
+          operation_launcher.compile(CompilationMode.Verify)
+          operation_launcher.compile(CompilationMode.Benchmark)
+
+        # Initialize verification and profiling results.
+        verification_result = 'Not run' if not verification_enabled else 'Failed'
+        runtime = -1.0
+
+        # Launch the operation dispatches for verification and profiling.
+        if verification_enabled:
+          verification_result = operation_launcher.verify(configuration)
+        if profiling_enabled:
+          runtime = operation_launcher.profile(configuration)
+
+        # Save and print the performance result.
+        if verification_enabled or profiling_enabled:
+          # Create and print a performance result.
+          result = PerformanceResult(operation_collection.operation,
+                                     configuration, verification_result,
+                                     runtime)
+          result.print()
+
+          # Append the performance result to the performance report.
+          perf_report.append_perf_result(result)
```

Comment:
Optional: I'll suggest putting this part into the `else` branch of the `compile_only` condition. It makes it easy to understand we don't do these steps if `compile_only==True`, instead of checking the line 116.

Reply:
Sounds good. Ultimately, I think we would like the compilation to be handled by BUILD files, but yes for now, I have made this change.